subprojects/core/src/main/java/org/gradle/process/internal/worker/DefaultWorkerProcess.java

```diff
@@ -165,11 +173,12 @@ private void doStart() {
                     throw UncheckedException.throwAsUncheckedException(e);
                 }
             }
-            if (processFailure != null) {
-                throw UncheckedException.throwAsUncheckedException(processFailure);
-            }
```

Comment:
Why is this moved into the `connection == null` case? I probably do not grasp the complete lifecycle. Are we certain that we will not miss any `processFailure` now?